packages/astro/src/core/build/plugins/plugin-content.ts

```diff
@@ -290,13 +355,65 @@ async function generateContentManifest(
 			);
 		}
 	}
+	
+	const [lockfiles, configs] = await Promise.all([
+		lockfilesHash(opts.settings.config.root),
+		configHash(opts.settings.config.root)
+	]);
+
+	manifest.lockfiles = lockfiles;
+	manifest.configs = configs;
 
 	await Promise.all(promises);
 	return manifest;
 }
 
-function checksum(data: string): string {
-	return createHash('sha1').update(data).digest('base64');
+async function pushBufferInto(fileURL: URL, buffers: Uint8Array[]) {
+	try {
+		const handle = await fsMod.promises.open(fileURL, 'r');
+		const data = await handle.readFile();
+		buffers.push(data);
+		await handle.close();
+	} catch {
+		// File doesn't exist, ignore
+	}
+}
+
+async function lockfilesHash(root: URL) {
+	// Order is important so don't change this.
+	const lockfiles = ['package-lock.json', 'pnpm-lock.yaml', 'yarn.lock', 'bun.lockb'];
```

Comment:
Why do we hash multiple lock files?

Reply:
It's possible to use multiple package managers in the same project. So we hash all of them in case this happens and one of them changes. ie you might use `pnpm` and I use `npm`, `pnpm-lock.yaml` is unchanged but `package-lock.json` is changed. This should result in an invalidation.

Reply:
Not even necessarily that you use multiple package managers, but that you have multiple in the same project. You might switch from `npm` to `pnpm`, but the package-lock.json is still in the project. We can't know which one is the "right" one to use, therefore hash them all.